glusterd2/config.go

```diff
@@ -117,7 +118,25 @@ func dumpConfigToLog() {
 	l.Debug("running with configuration")
 }
 
-func initConfig(confFile string) error {
+// initConfig intializes GD2 configuration from various sources.
+// The order of preference is,
+// - explicitly set configs using config.Set
+// - flags, if set
+// - environment variables
+// - config file
+// - defaults set using config.SetDefault
+// - flag defaults
+func initConfig() error {
+	// Use config given by flags
+	config.BindPFlags(flag.CommandLine)
+
+	// Allow config values from environment environment variables.
+	// All options settable from the command line are available to be set this way.
+	// The environment variable should be in uppercase, prefixed with "GD2" and have "-" replaced by "_" to be used.
+	config.AutomaticEnv()
```

Comment:
May be this doesn't matter - but I assume `AutomaticEnv` should come after setting the following two - `SetEnvPrefix` and `SetEnvKeyReplacer`, at least conceptually

Reply:
This doesn't matter till a `config.Get` is done. But for the sake of conceptual consistency, I'll change the order.